st2api/setup.py

```diff
@@ -14,27 +14,43 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-try:
-    from setuptools import setup, find_packages
-except ImportError:
-    from ez_setup import use_setuptools
-    use_setuptools()
-    from setuptools import setup, find_packages
+import re
+import os.path
+from pip.req import parse_requirements
+from setuptools import setup, find_packages
+
+
+def fetch_requirements():
+    links = []
+    reqs = []
+    for req in parse_requirements('requirements.txt', session=False):
+        if req.link:
+            links.append(str(req.link))
+        reqs.append(str(req.req))
+    return (reqs, links)
+
+
+current_dir = os.path.dirname(os.path.realpath(__file__))
+version_file = os.path.join(current_dir, '../st2client/st2client/__init__.py')
+with open(version_file, 'r') as f:
```

Comment:
I've added top `st2_version` file, which is being populated for each component also if it can be found. This is done by make, so by the type setup.py is run, all is nice file can be cat.